### ISSUE TITLE
Pretrained model output

### ISSUE THREAD

**Issue by ermolushka** [fangyiyu/GNPassGAN#2]
Hi. I tried to run pretrained model to generate samples: it looks like it creates non-relevant passwords which don't look like real one. Examples:

```
   BJ}I)B.
EVTN}
     q^^unkqX
^wtEB
     o#o)E^
K
 eVw^-s^N.N
6E-^!BunkoEJtk
BN6^N)E}^!^.
^eME*.^NNunk^!
TunkqNM^EJe&t^
KB)Te9JB>M)-
wM!
   !NEBunk)LX
kNEE--e^eM^!
EM)NM&B\MwoM
```

In comparison, when I tried pre-trained model of PassGAN, it returned more meaningful results. Could you please assist, why it happened?

**Comment by ermolushka** [comment]
@fangyiyu

**Comment by fangyiyu** [comment]
Thanks for being interested in our work. 

How to compare GNPassGAN and PassGAN is to generate the same number of passwords using the two techniques, say 10,000,000, then count how many real passwords in the test set exist in these 10,000,000 passwords. The matching ratio increases dramatically once the magnitude of passwords generated increases. When 100,000,000 passwords are generated by GNPassGAN, 12.647% of the passwords in the test set exist in the generated passwords, then the remaining generated passwords may look real or unreal.

More information can be found in our paper: GNPassGAN: Improved Generative Adversarial Networks For Trawling Offline
Password Guessing.

Hope it helps!
